lib/RGBot.js

```diff
@@ -294,145 +551,297 @@ const RGBot = class {
      */
     async handlePath(pathFunc, options = {}) {
         const interval = options.interval || 5000;
-        let previousPosition = this.#bot.entity.position;
-        let wasActive = true;
+        let previousPosition = this.position();
         let stuck = false;
 
         const checkPosition = () => {
-            let currentPosition = this.#bot.entity.position;
-            let isActive = this.#bot.pathfinder.isMining() || this.#bot.pathfinder.isBuilding();
-            if (currentPosition.equals(previousPosition, 0.005) && !wasActive && !isActive) {
-                // if the bot hasn't moved or performed other actions then we are stuck
-                // stop pathfinder and remove its current goal
-                console.error(`handlePath: Bot is stuck. Stopping current path.`);
-                stuck = true;
-                this.#bot.pathfinder.setGoal(null);
-                this.#bot.pathfinder.stop();
-            } else {
-                previousPosition = currentPosition;
-                wasActive = isActive;
+            try {
+                let currentPosition = this.#bot.entity.position;
+                if (currentPosition.distanceTo(previousPosition) < 2 && !this.#bot.pathfinder.isMining() && !this.isLookingInContainer && !this.isCrafting) {
+                    // if the bot hasn't moved or performed other actions then we are stuck
+                    // stop pathfinder and remove its current goal
+                    this.#logWarn(`!!!!!!!!!!!!!! Bot is stuck at ${this.vecToString(currentPosition)}. This normally means that your bot is glitch stuck inside a block, or that your bot code does not have a way to move it from its current location.  You can manually unstick your bot by digging it free, or by updating your bot code to handle the case it is stuck in.  Also visit https://play.regression.gg/documentation/faq#my-bot-gets-stuck-while-pathfinding`);
+                    stuck = true;
+                    this.#bot.pathfinder.setGoal(null);
+                } else {
+                    previousPosition = currentPosition;
+                }
+            } catch (ex) {
+                // don't log anything if we get GoalChanged here, that is expected when stopping a stuck bot
             }
         }
 
         const timer = setInterval(checkPosition, interval);
         try {
             await pathFunc();
+        } catch (err) {
+            if (stuck) {
+                // don't log anything if we get GoalChanged here, that is expected when stopping a stuck bot
```

Comment:
Confused by this comment since we are catching goalChanged events for the same reason above

Reply:
Depending on the timing, you can get the exception on either spot.

I think now that I completely removed the call to `.stop()`, it may be impossible to get the message in the `checkPosition` `try/catch`, but I'm inclined to leave that try catch there as I wouldn't ever want our position check to blow up their action accidentally.